### ISSUE TITLE
[Feature] Cloud server support (Yunohost, UCS, Cloudron)

### ISSUE THREAD

**Issue by hermann-san** [turtl/tracker#351]
Open Source self hosting systems like  Yunohost, UCS Univention or Sandstorm.io  are becoming increasingly popular because they are easy to use by regular users. 
It would be great if Turtl would support Yunohost or UCS.  That would greatly increase the visibility of Turtl to Open Source cloud community. I know that in Germany many schools are moving and looking into UCS because it's easy to administer and more cost-effective to run then e.g. Microsoft Exchange - and of course Open Source apps are used. 

App Catalog
https://www.univention.com/products/univention-app-center/app-catalog/ (e.g. Media Wiki, Kopano, Etherpad, Mattermost (Matrix/Riot), Nextcloud, Rocket-Chat, ....
https://yunohost.org/#/apps    e.g. DokuWiki, Etherpad, Wallabag, Kanboard and many more.

Here it is explained how you could get apps added to UCS
https://www.univention.com/products/univention-app-center/for-solution-providers/ 
https://yunohost.org/#/packaging_apps

I don't know why it's so hard to get a good Evernote alternative done. Turtle seems to me currently best and most promising Self-hosting Evernote alternative on the Open Source scene. It would just be good if it were a little easier to install and maintain. This is were the cloud-server kicks-in. Since Evennote is so popular there must also be a market where people would also pay for an open app . People and companies pay for hosted Open Source apps like  Mattermost , Rocket Chat, Taiga and other).

**Comment by fbartels** [comment]
In case there is a docker-compose.yml with the whole Turtl stack a UCS could be quite easy to get started. UCS is using the compose file to handle multi container apps in their appcenter.

Another platform to provide Turtl for could be https://cloudron.io/. Cloudron apps are single container apps, so the same container would need to provide both the back and frontend (database etc is provided by Cloudron).

I have already built apps for both platforms and would be open to collaborate on this.

**Comment by hermann-san** [comment]
@fbartels  a Docker image seems to be available for Turtl, but unfortunately I'm not a Linux expert
https://github.com/turtl/server

**Comment by fbartels** [comment]
Yes, I am aware of that repo, but that only seems to be the backend an its missing a web frontend. overall there does not seem to be an installable web frontend for turtl (only mobile and desktop apps), which certainly reduces its usefulness.

For Yunohost I also found the following:
https://github.com/YunoHost-Apps/Turtl_ynh
https://forum.yunohost.org/t/bounty-10-usd-over-paypal-for-someone-to-package-turtl-server/7063

Nevertheless if one of the Turtl contributors/developers would be open to collaborate on an app I would still be available.

**Comment by hermann-san** [comment]
I've also seen Turtle for Yunohost, but it was marked as "not working" (I can't remember were exactly I've seen it). Also I can't install it from within the Yunohost installer (It's not there) and it's not listed as working on the Yunohost Dashboard here
https://github.com/turtl/tracker/issues/351?email_source=notifications&email_token=AB2S5JSYNRDF6VH4LBPMZELQ3B4KVA5CNFSM4J4UYUR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHY4M2Q#issuecomment-569493098

Web front end is missing, correct - but personally,  I don't need it urgently since I have a desktop app and a smartphone app.   The problem is that there is simply no proper, easy to use, self hosted, open source alternative to Evernote/Turle that I know of. 

Thanks for offering your help to get Turtl to Yunohost.

**Comment by juju4** [comment]
IMHO, It would be great to be able to use nextcloud to store notes, ideally encrypted or not.
non-encrypted can be edited directly as text/markdown file
encrypted ones would need a nextcloud app to be edited in nextcloud web interface.
